### ISSUE TITLE
Proxy support

### ISSUE THREAD

**Issue by shinoshu** [mattermost/mmctl#187]
What should I do for proxy settings?

**Comment by isacikgoz** [comment]
Could you elaborate your problem a bit more? I'm going to close this for cleaning the queue. @shinoshu feel free to open this issue again with a detailed explanation.

**Comment by nealey** [comment]
Maybe I can help!

I am behind a corporate firewall and must use an http proxy for outbound connections. Normally this is specified with the `HTTP_PROXY` or `HTTPS_PROXY` environment variables, but `mmctl` doesn't seem to look at these.

```sh
$ docker run --rm -e HTTP_PROXY=http://proxyout.lanl.gov:8080/ -e HTTPS_PROXY=http://proxyout.lanl.gov:8080/ -v /home/neale/.config/mmctl:/mattermost/.config/mmctl:ro mattermost/mattermost-team-edition mmctl channel list foundry --format=json
Error: Get "https://mattermost.cyberfire.training/api/v4/users/me": dial tcp 3.143.159.223:443: connect: connection refused
```

It should not be calling `dial()` on the IP address of `mattermost.cyberfire.training`, but rather, it should connect to `proxyout.lanl.gov` and send an HTTP `CONNECT` command to establish the HTTPS connection.

I believe Go's [ProxyFromEnvironment](https://pkg.go.dev/net/http#ProxyFromEnvironment) call can be used to get environment information about this stuff. `DefaultClient` will also honor it, but you're probably doing more advanced stuff that precludes that.

**Comment by isacikgoz** [comment]
@shinoshu @nealey the issue is now truly resolved with #579! It seems like I missed out the comment after the issue is closed. Sorry for the delay!

**Comment by nealey** [comment]
Oh, gosh, I didn't realize I had commented on a closed ticket! Thanks for finding it anyway, and also thanks for demonstrating to me how to set up an http proxy in a Go unit test ;)
